### ISSUE TITLE
OpenBSD: reslovd prioritizes dhclient route proposals over tailscale nameserver

### ISSUE THREAD

**Issue by sauterp** [tailscale/tailscale#3722]
I am using tailscale on OpenBSD 7.x on a notebook with a WiFi card and DHCP. DNS resolution didn't work reliably for tailscale hosts on my LAN after installing the [tailscale port](https://cvsweb.openbsd.org/cgi-bin/cvsweb/ports/net/tailscale/).
Even if I add the tailscale nameserver manually to `/etc/resolv.conf` like this:
```
nameserver 100.100.100.100
nameserver 192.168.1.1 # resolvd: iwn0
```

resolvd will later flip the line order, thus sending lookups to my home router first. If hosts change their IP address I cannot reach them anymore via their hostnames, since the nameserver of my router doesn't update quick enough.

I was able to solve this problem by enabling unwind.

`rcctl enable unwind`

With a configuration file `/etc/unwind.conf`:

```
forwarder { 100.100.100.100 }
preference forwarder
```

resolvd detects the unwind service and will update `/etc/resolv.conf` accordingly:

`nameserver 127.0.0.1 # resolvd: unwind`

This appears to solve the problem. To provide the same out-of-the-box experience as on other OSes however, we cannot package this solution, as it would force users to enable unwind.

Is there a better way I'm not aware of maybe?

Thank you.

related issues:
#815
#375

**Comment by qbit** [comment]
You are using the version in ports? I added support for working through resolvd but it isn’t in the ports version yet.

**Comment by sauterp** [comment]
I'm not a seasoned OpenBSD user, maybe I confused some vocabulary. I installed tailscale via `pkg_add`.

And yes, I'd be happy to test a new version of the package or port.

**Comment by DentonGentry** [comment]
Fixed by https://github.com/tailscale/tailscale/pull/2789, which was included in Tailscale 1.20.2.
